confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceConverter.java

```diff
@@ -52,25 +60,27 @@
 import org.xwiki.rendering.listener.reference.ResourceType;
 import org.xwiki.rendering.listener.reference.UserResourceReference;
 
-import javax.inject.Inject;
-import javax.inject.Named;
-import javax.inject.Provider;
-import javax.inject.Singleton;
-import java.util.Collections;
-import java.util.Map;
-import java.util.regex.Pattern;
+import com.xpn.xwiki.XWikiContext;
 
 import static org.xwiki.contrib.confluence.filter.input.ConfluenceXMLPackage.KEY_GROUP_EXTERNAL_ID;
 import static org.xwiki.contrib.confluence.filter.input.ConfluenceXMLPackage.KEY_GROUP_NAME;
 
 /**
- * Default implementation of ConfluenceConverter.
+ * Default implementation of ConfluenceFilterReferenceConverter.
+ *
+ * Note: While the implementation itself is not deprecated, the ConfluenceConverter role is deprecated and will
+ * eventually go away. We give a grace period to let projects switch to {@link ConfluenceFilterReferenceConverter},
+ * ideally we should get rid of this deprecated role early 2026. We use the deprecated annotations to make the IDEs
+ * warn against using it.
+ *
  * @version $Id$
  * @since 9.26.0
+ * @deprecated since 9.83.0, use {@link ConfluenceFilterReferenceConverter} instead
  */
-@Component(roles = ConfluenceConverter.class)
+@Component (roles = {ConfluenceFilterReferenceConverter.class, ConfluenceConverter.class})
 @Singleton
-public class ConfluenceConverter implements ConfluenceReferenceConverter
+@Deprecated (since = "9.83.0")
```

Comment:
Do you need to keep this component since it's internal ? Don't want to fix all the code that calls it just yet ?